### PR TITLE
Added redspot tests for Access Control contract

### PR DESCRIPTION
**Pull request by PierreOssun**

### CODE REVIEW

File: tests/access-control.tests.ts [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+  })
+
+  it('ACCESS CONTROL - should grant role', async () => {
+    const {
+      query,
+      tx,
+      accounts: [alice]
+    } = await setup()
+
+    await tx.grantRole(Roles.Minter, alice.address)
```
Comment (xgreenx) [inline tests/access-control.tests.ts:175]: Could you verify that alice didn't have the role before please?

File: tests/access-control.tests.ts [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+      contract,
+      accounts: [alice]
+    } = await setup()
+
+    await tx.grantRole(Roles.Minter, alice.address)
+
+    await expect(fromSigner(contract, defaultSigner.address).tx.renounceRole(Roles.Minter, alice.address)).to.eventually.be.rejected
+  })
+
+  it('ACCESS CONTROL - should reject burn if no minter role', async () => {
```
Comment (xgreenx) [inline tests/access-control.tests.ts:247]: Could you also create a test where the admin will provide Minter role to Alice, and Alice calls mint please?=)

File: tests/helpers.ts [hunk 3]
```diff
@@ -37,20 +47,8 @@ export const setupContract = async (name, constructor, ...args) => {
   }
 }
 
-// It removes prefix from the function and adds only name of method like a function
```
Comment (xgreenx) [inline tests/helpers.ts:40]: You forgot to copy the comment)

File: contracts/access/access-control/impls.rs [hunk 4]
```diff
@@ -89,6 +89,11 @@ pub trait AccessControl: AccessControlStorage {
 
     // Internal functions
 
+    fn _init(&mut self) {
```
Comment (xgreenx) [inline contracts/access/access-control/impls.rs:92]: Let's call it `_init_with_caller`

File: tests/access-control.tests.ts [hunk 5]
```diff
@@ -0,0 +1,257 @@
+import { consts, Roles } from './constants'
+import {bnArg, expect, fromSigner, setupContract} from './helpers'
+
+describe('MY_ACCESS_CONTROL', () => {
+  async function setup() {
+    return setupContract('my_access_control', 'new')
+  }
+
+  it('ERC 721 - mint works', async () => {
```
Comment (sventime) [inline tests/access-control.tests.ts:9]: Let's move ERC 721 tests to erc721.tests.ts file

File: tests/access-control.tests.ts [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+    expect(owner.output).to.equal(sender.address)
+
+    await expect(contract.tx.approve(alice.address, bnArg(0)))
+      .to.emit(contract, 'Approval').withArgs(sender.address, alice.address, bnArg(0))
+
+    await expect(fromSigner(contract, alice.address).tx.transferFrom(sender.address, bob.address, bnArg(0)))
+      .to.emit(contract, 'Transfer').withArgs(sender.address, bob.address, bnArg(0))
+
+    const ownerOf = await query.ownerOf(bnArg(0))
+    expect(ownerOf.output).to.equal(bob.address)
```
Comment (sventime) [inline tests/access-control.tests.ts:63]: You can create to.have.output(...) matcher and do all checks in one row:
```
chai.use((chai) => {
  chai.Assertion.addMethod('output', async function (param, message) {
    await new chai.Assertion(this._obj).to.eventually.have.property('output').to.equal(param, message)
  })
})
export const { expect } = chai
```

File: tests/access-control.tests.ts (hunk 5 above)
Comment (xgreenx) [reply tests/access-control.tests.ts:9]: I prefer to not create a lot of files with suffixes. I think it better to have one file for one example. Inside of test you can you several description sections to separate tests for Erc71, AccessControl, and Both.

File: tests/access-control.tests.ts (hunk 2 above)
Comment (PierreOssun) [reply tests/access-control.tests.ts:270]: I updated _ACCESS CONTROL - only minter role is allowed to mint_ for it

File: tests/access-control.tests.ts (hunk 6 above)
Comment (PierreOssun) [reply tests/access-control.tests.ts]: Good point. thanks

**Review by xgreenx — APPROVED** [review]

**Pull request merged**